### PR TITLE
enhance gcw tool with search

### PR DESCRIPTION
**Pull request by t-m-z**
first idea for review and discussion

### CODE REVIEW

**Comment by t-m-z** [comment]
closes also #388

**Comment by S-Man42** [comment]
Please fix the registry conflicts.

File: lib/widgets/common/gcw_tool.dart [hunk 1]
```diff
@@ -18,6 +19,21 @@ enum ToolCategory {
   SYMBOL_TABLES
 }
 
+final SearchBlackList = {
+  'der',
```
Comment (S-Man42) [inline lib/widgets/common/gcw_tool.dart:23]: ein, eine, un, une, d', l', le, la, of,

**Comment by t-m-z** [comment]
merge problems solved
registry checked
blacklist enhanced

File: lib/widgets/common/gcw_tool.dart [hunk 2]
```diff
@@ -18,6 +19,28 @@ enum ToolCategory {
...
+  'three',
+  'of',
+  'from',
+  'and',
+  'deux',
+  'trois',
+  'de',
+  "l'",
+  'le',
+  'la'
```
Comment (S-Man42) [inline lib/widgets/common/gcw_tool.dart:41]: Was it intended that you left out: ein, eine, du (== french de + le), un , une, a (== french à and english a)? One can add "one" as well...

File: lib/widgets/common/gcw_tool.dart [hunk 3]
```diff
@@ -100,19 +126,47 @@ class _GCWToolState extends State<GCWTool> {
...
-
     String url = '';
 
+    // add button with url for searching knowledge base with toolName
+    final Locale appLocale = Localizations.localeOf(context);
+    if (isLocaleSupported(appLocale))
+      url = 'https://blog.gcwizard.net/manual/' +
+          Localizations.localeOf(context).toString() +
+          '/search/' +
+          widget.toolName;
```
Comment (S-Man42) [inline lib/widgets/common/gcw_tool.dart:137]: I guess you need to escape this, correct? Or does "Affine Cipher" with spaces work?

File: lib/widgets/common/gcw_tool.dart [hunk 4]
```diff
@@ -100,19 +126,47 @@ class _GCWToolState extends State<GCWTool> {
...
+    else // fallback to en if unsupported locale
+      url = 'https://blog.gcwizard.net/manual/' +
+          'en' +
+          '/search/' +
+          widget.toolName;
+    SearchBlackList.forEach((element) {
```
Comment (S-Man42) [inline lib/widgets/common/gcw_tool.dart:143]: It shouldn't be case sensitive... Moreover in cases of french d' and l', there's no space following.

File: lib/widgets/common/gcw_tool.dart (hunk 3 above)
Comment (t-m-z) [reply lib/widgets/common/gcw_tool.dart:137]: "Affine cipher" with spaces work

File: lib/widgets/common/gcw_tool.dart (hunk 2 above)
Comment (t-m-z) [reply lib/widgets/common/gcw_tool.dart:41]: the blacklist is a living document - so this was the first rough guess

File: lib/widgets/common/gcw_tool.dart (hunk 4 above)
Comment (t-m-z) [reply lib/widgets/common/gcw_tool.dart]: Because we do a full text search the removals of Diacritics should be avoided

**Comment by t-m-z** [comment]
unfortunately I cannot test the changes due to an error message I do not understand:
A problem was found with the configuration of task ':app:createDebugCompatibleScreenManifests' (type 'CompatibleScreensManifest').
File 'D:\Programmierung\GCWizard\build\app\intermediates\apk_list\debug\apk-list.gson' specified for property 'apkList' does not exist.

I saw an update of gradle to 6.3.8 and no this occurs.
I have already deleted the .gradle but no improvement.

**Pull request merged**